### PR TITLE
Implement distribution spec and contractual build plan RFCs

### PR DESCRIPTION
**Pull request by sclevine**
Context:
https://github.com/buildpack/rfcs/pull/11
https://github.com/buildpack/rfcs/pull/12

Spec PR for build plan: https://github.com/buildpack/spec/pull/52
Spec PR for distribution spec: https://github.com/buildpack/spec/pull/53

### CODE REVIEW

**Comment by sclevine** [comment]
@jkutner @hone @nebhale this is ready to test out, but you'll need to manually construct the builder

File: Makefile [hunk 1]
```diff
@@ -1,6 +1,7 @@
-# Go parameters
-GOCMD?=go
-GOENV=GO111MODULE=on GOOS=linux GOARCH=amd64 CGO_ENABLED=0
+export GO111MODULE = on
+
+GOCMD=go
```
Comment (ameyer-pivotal) [inline Makefile]: We should keep the `?` so that it's overridable from the command line (we often swap out to `richgo`)

Comment (sclevine) [reply Makefile]: Sorry, yep, keep it.

**Comment by sclevine** [comment]
Updated to match https://github.com/buildpack/rfcs/pull/15.

**Comment by ekcasey** [comment]
@sclevine we we rebase this it would be nice to squash the small wip commits into a couple larger commit to improve the readability of the history

File: .travis.yml [hunk 2]
```diff
@@ -1,22 +1,13 @@
...
-    go_import_path: github.com/buildpack/lifecycle
-    script: |
-      make test
-
+language: go
+go:
+- 1.12.x
+before_script:
+- mkdir /tmp/yj
+- wget https://github.com/sclevine/yj/releases/download/v4.0.0/yj-linux -O /tmp/yj/yj
```
[the remark this answers is not in the document]
Comment (sclevine) [reply .travis.yml]: Definitely

**Pull request merged**